var/spack/repos/builtin/packages/zoltan/package.py

```diff
@@ -0,0 +1,54 @@
+from spack import *
+
+class Zoltan(Package):
+    """The Zoltan library is a toolkit of parallel combinatorial algorithms for
+       parallel, unstructured, and/or adaptive scientific applications.  Zoltan's
+       largest component is a suite of dynamic load-balancing and paritioning
+       algorithms that increase applications' parallel performance by reducing
+       idle time.  Zoltan also has graph coloring and graph ordering algorithms,
+       which are useful in task schedulers and parallel preconditioners."""
+
+    homepage = "http://www.cs.sandia.gov/zoltan"
+    base_url = "http://www.cs.sandia.gov/~kddevin/Zoltan_Distributions"
+
+    version('3.83', '1ff1bc93f91e12f2c533ddb01f2c095f')
```

Comment:
Looks like this is how they really write versions for Zoltan, just note that Spack will consider `3.83` higher than `3.8.4`.  As long as they pick one and do not mix `x.y.z` and `x.yz`, things in Spack should behave the way you expect.